null_test.go

```diff
@@ -0,0 +1,26 @@
+package logr
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestNull(t *testing.T) {
+	l := Null()
+	l.WithName("null").WithValues("z", 5).Info("Hello world")
```

Comment:
This isn't really a test, though - if the test passes logs won't be shown.  I'm a fan of tests when they add value, I am not sure this is worth the code, though?

Reply:
I added this to ensure that the implementation does not panic instead of returning subloggers, which is my only functional requirement for it.